src/middleware/navigation/update/has.beneficial.owners.or.managing.officers.update.middleware.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response, NextFunction } from 'express';
+
+import { logger } from '../../../utils/logger';
+import { UPDATE_BENEFICIAL_OWNER_TYPE_URL } from '../../../config';
+import { getApplicationData } from "../../../utils/application.data";
+import { checkBOsOrMOsDetailsEnteredUpdate, NavigationErrorMessage } from '../check.condition';
+
+export const hasBOsOrMOsUpdate = (req: Request, res: Response, next: NextFunction): void => {
+  try {
+    if ( !checkBOsOrMOsDetailsEnteredUpdate(getApplicationData(req.session)) ) {
+      logger.infoRequest(req, NavigationErrorMessage);
+      return res.redirect(UPDATE_BENEFICIAL_OWNER_TYPE_URL);
```

Comment:
Did this redirect location come from product? I've a faint memory of product mentioning navigation check failures should take us to the beginning of the journey. Worth double checking :)

Reply:
Following discussion with Amelia Ward it was agreed to send user back to BO/MO type page. In register journey on the check-your-answers navigation, there is a check for BOs and MOs which sends the user back to the sold land filter page- this page does not exist in update so we agreed at the time it should be changed to the BO/MO page. However upon reflection this should probably be the secure update filter page to fit in line with the other validation checks. I will  confirm this with product

Reply:
This has been changed to send the user back to the start of the journey (secure update filter)